lib/semian/mysql2.rb

```diff
@@ -53,11 +53,19 @@ def semian_identifier
     end
 
     def query(*args)
-      acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      if query_whitelisted?(*args)
+        raw_query(*args)
+      else
+        acquire_semian_resource(adapter: :mysql, scope: :query) { raw_query(*args) }
+      end
     end
 
     private
 
+    def query_whitelisted?(sql, *)
+      sql =~ /\A\s*ROLLBACK/i
```

Comment:
is rubby smart enough to not instantiate a new regexp every time? if not make it a constant